### PR TITLE
[APO-1189] Add WebSearchNode Code Generation Support

### PR DESCRIPTION
**Pull request by VJagiasi**
[Ticket](https://linear.app/vellum/issue/APO-1189/add-support-for-code-generation-for-web-search-node)

## Rationale
  Implements code generation support for WebSearchNode to enable users to pull and generate Python workflow code containing web search functionality, completing the WebSearchNode implementation chain.

 ## Changes
  - **Core Configuration**: Added WebSearchNode to NODE_ATTRIBUTES in `constants.ts` with attributes: `query`, `api_key`, `num_results`, `location`
  - **Test Infrastructure**: Created `webSearchNodeFactory` for generating test node data
  - **Comprehensive Test Suite**: 8 focused test cases covering:
    - Basic node and display file generation
    - Workflow input integration
    - Different value types (NODE_OUTPUT, DICTIONARY_REFERENCE)
    - Parameter validation and edge cases
   
## Details
  - WebSearchNode uses simple attribute configuration (no complex AttributeType enums like ToolCallingNode)
  - All attributes processed through standard WorkflowValueDescriptor handling
  - Generated code imports from `vellum.workflows.nodes.displayable.web_search_node`
  - VELLUM_SECRET attributes use `vellumSecretName` field for API key references

### CODE REVIEW

File: ee/codegen/src/__test__/nodes/__snapshots__/web-search-node.test.ts.snap [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+)
+from vellum.workflows.references import EnvironmentVariableReference
+
+from ..inputs import Inputs
+
+
+class WebSearchNode(BaseWebSearchNode):
+    query = {
+        "search_term": Inputs.query,
+    }
```
Comment (dvargasfuertes) [inline ee/codegen/src/__test__/nodes/__snapshots__/web-search-node.test.ts.snap]: [q] Is this a valid shape that our Node / SerpAPI supports?

**Review by dvargas92495 — APPROVED** [review]

**Review by vincent0426 — APPROVED** [review]

File: ee/codegen/src/__test__/nodes/__snapshots__/web-search-node.test.ts.snap (hunk 1 above)
Comment (VJagiasi) [reply ee/codegen/src/__test__/nodes/__snapshots__/web-search-node.test.ts.snap]: you're right - that dictionary structure isn't valid for our WebSearchNode

dug into this and confirmed:
  1. The WebSearchNode expects query to be a string (validates with isinstance(self.query, str))
  2. SerpAPI's q parameter only accepts strings, not objects
  3. We have no dict-to-string processing in the node implementation

That test case was generating invalid Python code that would fail at runtime. removed the DICTIONARY_REFERENCE test case since it represents an invalid scenario
